src/taskpane/pages/dashboard.tsx

```diff
@@ -120,21 +151,38 @@ function Dashboard({ citeSupport }: DashboardProps): ReactElement {
   };
 
   const isCitationEdited = (): boolean => {
-    return arraysEqual(
-      checkedItems.map((i) => i.id),
-      itemsIDsInSelectedCitation.current
+    return (
+      JSON.stringify(checkedItems.map((citation) => ({ ...citation }))) ===
+      JSON.stringify(itemsIDsInSelectedCitation.current)
     );
   };
 
+  const updateCitationMetaData = (citation: CitationItem) => {
+    setItems((currentItems) => {
+      return currentItems.map((item) => {
+        if (item.id === citation.id) {
+          return {
+            ...item,
+            label: citation.label,
+            locator: citation.locator,
+            prefix: citation.prefix,
+            suffix: citation.suffix,
+          };
+        }
+        return item;
+      });
+    });
+  };
+
   const getSelectedCitation = useCallback(async (): Promise<void> => {
-    const getItemsIDInCitation =
+    const getItemsMetaDataInCitation =
```

Comment:
remove the `get` prefix (variables usually are called like the object it represents...so if you name something `getXYZ` then one thinks it is a method that when invoked gives xyz back).